src/main/java/org/springframework/data/couchbase/core/ReactiveRemoveByIdOperationSupport.java

```diff
@@ -81,28 +83,36 @@ private RemoveOptions buildRemoveOptions() {
 			} else if (durabilityLevel != DurabilityLevel.NONE) {
 				options.durability(durabilityLevel);
 			}
+			if (cas != null) {
```

Comment:
I see the boxing and null, but it could just check if != 0 ?

Reply:
This comes from a comment made by Mark Paluch on another change-set regarding Expiry(I think). It's not quite the same for CAS, but I wanted to keep the same pattern/semantics - that null means "never been set/not specified", while an Object (including an object with value 0) means that it has been set.

https://github.com/spring-projects/spring-data-couchbase/pull/292#pullrequestreview-549850646

In other words - something like this -  That if it was set to non-Zero by an annotation, then there was no way to override (unset) it with the fluent call because it was already non-Duration.zero and the fluent call had to set it to Duration.zero, but Duration.zero (from the fluent call or otherwise) is treated as "never been set", so the value from the annotation would not be overridden. So null would have the semantics of "never been set", and Duration.zero has the semantics of "been set to zero".